pkg/scheduler/scheduler.go

```diff
@@ -135,17 +135,17 @@ func (s *Scheduler) scheduleOnce(ctx context.Context, worker int) {
 	// Retrieve the placement object (either ClusterResourcePlacement or ResourcePlacement).
 	placement, err := controller.FetchPlacementFromKey(ctx, s.client, placementName)
 	if err != nil {
```

Comment:
should we check if the error is unexpected and then not requeue the key?

Reply:
I think Not Requeuing the key is important to prevent the outage. In the worst case, if regression happens, the queue may be full of invalid keys, which will keep our agent busy and cause service unavailable.